### ISSUE TITLE
Add `null_count` for LazyFrames

### ISSUE THREAD

**Issue by lucazanna** [pola-rs/polars#8828]
### Problem description

I am converting some code from using DataFrames to using LazyFrames, and I realize that `null_count` is not available for Lazyframes.
It's available for DataFrames, Series and Expressions but not LazyFrames.

Is there a reason for it?

A different way to achieve the same is by using null_count with an expression, then collecting the expression. However, having the LazyFrame method makes for a cleaner / more aligned code.

Your thoughts?

**Comment by cmdlineluser** [comment]
A collect needs to happen in order to calculate the count. 

If a `.null_count` method existed - you would have a "hidden" collect going on - which doesn't seem like something that should be in a lazy api?

**Comment by lucazanna** [comment]
hi @cmdlineluser ,

I would not include a hidden collect in `null_count`. I would add it explicitly.

here's an example:
```python
# dataframe null count all columns -> works
df.null_count()

# lazyframe null count all columns -> doesn't work today
lf.null_count().collect()

# dataframe null count one column -> works
df.select(
    pl.col('col1').null_count()
)

# lazyframe null count one column -> works
lf.select(
    pl.col('col1').null_count()
).collect()
```
my proposal is to allow the second one to work.

the alternative for the second case is to do a `select` and a `pl.all`. That works, but it would seem consistent to allow a null_count for lazyframes.

you see an issue with the second example?

**Comment by ritchie46** [comment]
My reservation here is that I don't see the utility here for anything other than EDA.

It implicitly selects all columns, meaning that you fully materialize. That doesn't fit the intent of lazy where we only select columns users really need.

**Comment by lucazanna** [comment]
@ritchie46 : I understand about materializing everything.

the main value I see in adding this is consistency. 
When I see people learning Polars, a good rule of thumb they learn is that most dataframes transformations will work on lazyframes. 

The other advantage of allowing null_count on Lazyframes would be for streaming. 
For example, if I have a high volume of data arriving every day and I want to ensure there are no null values, then I can use this.

**Comment by ritchie46** [comment]
Yes, but exactly in streaming you shouldn't select all colums of you don't need it. 

Some methods are only on eager for EDA and materialization reasons. We can perfectly explain that.

**Comment by alexander-beedie** [comment]
Hmm... The same could be said about the other statistics that we already support on `LazyFrame`, such as `median`, `mean`, `max`, `min`, `sum`, etc. Given that we already have those, I'm not sure there's any reason to single out `null`_count` here.

Anyway, it's a simple enough addition that fits with the existing/similar methods already found on LazyFrame, so it seems fairly harmless; or maybe we could make a stand and actively remove the other methods, heh 😄

I made a small PR that adds it in the same way (and in the same place) as the aforementioned methods; have a look and see what you think. My own thoughts are that if we're going to support those other methods, it would be consistent if we also support this.

https://github.com/pola-rs/polars/pull/8837

**Comment by cmdlineluser** [comment]
> `lf.null_count().collect()`

@lucazanna D'oh, I completely missed the obvious there, apologies. 😭
